src/querier.rs

```diff
@@ -53,19 +54,19 @@ impl Querier for GoQuerier {
             )
         };
         unsafe {
-            go_result.into_ffi_result(err, default)?;
+            if let Err(err) = go_result.into_ffi_result(err, default) {
+                return (Err(err), gas_info);
+            }
         }
 
         let bin_result = unsafe { result_buf.consume() };
-        match serde_json::from_slice(&bin_result) {
-            Ok(system_result) => Ok((system_result, used_gas)),
-            Err(e) => Ok((
-                Err(SystemError::InvalidResponse {
-                    error: format!("Parsing Go response: {}", e),
-                    response: Binary(bin_result),
-                }),
-                used_gas,
-            )),
-        }
+        let result = match serde_json::from_slice(&bin_result) {
+            Ok(system_result) => Ok(system_result),
+            Err(e) => Ok(Err(SystemError::InvalidResponse {
+                error: format!("Parsing Go response: {}", e),
+                response: Binary(bin_result),
+            })),
+        };
```

Comment:
I think this is equivalent, if you want to avoid the `match` expression:
```rust
let result = serde_json::from_slice(&bin_result).or_else(|e| {
  Err(SystemError::InvalidResponse {
    error: format!("Parsing Go response: {}", e),
    response: Binary(bin_result)
  })
});
```

Reply:
Done in https://github.com/CosmWasm/go-cosmwasm/pull/123/commits/da29355864e5d917716e379be9d6f5ef24238497, thank you